File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        output = 0
        numeralMap = {
            "M" : 1000,
            "D" : 500,
            "C" : 100,
            "L" : 50,
            "X" : 10,
            "V" : 5,
            "I" : 1
        }
        i = len(s) - 1
        prevNum = 0

        while i >= 0:
            currNum = numeralMap[s[i]]
            if currNum < prevNum:
                output -= currNum
            else:
                output += currNum
            prevNum = currNum
            i -= 1
        
        return output

File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {

        int res = 0;

        unordered_map<char, int> roman = {
            {'M' , 1000},
            {'D' , 500},
            {'C' , 100},
            {'L' , 50},
            {'X' , 10},
            {'V' , 5},
            {'I' , 1}
        };

        int prevNum = 0;

        for (int i = s.size() - 1; i >= 0; --i) {
            int currNum = roman[s[i]];
            if (currNum < prevNum) res -= currNum;
            else res += currNum;
            prevNum = currNum;
        };
        return res;
    };
};